ingestion/src/metadata/ingestion/source/metadata/atlas.py

```diff
@@ -126,20 +124,38 @@ def prepare(self):
         """
 
     def next_record(self):
-        for key in ENTITY_TYPES["Table"]:
-            self.service = self.get_database_service()
-            self.tables[key] = self.atlas_client.list_entities(entity_type=key)
+        for service in self.service_connection.databaseServiceName or []:
 
-        for key in ENTITY_TYPES.get("Topic", []):
-            self.message_service = self.get_message_service()
-            self.topics[key] = self.atlas_client.list_entities(entity_type=key)
+            try:
+                create_service = self.metadata.get_by_name(
+                    entity_type=DatabaseService, fqn_search_string=service
+                )
+                for key in ENTITY_TYPES["Table"]:
+                    self.service = create_service
+                    self.tables[key] = self.atlas_client.list_entities(entity_type=key)
+                    if self.tables:
+                        for key in self.tables:
+                            yield from self._parse_table_entity(key, self.tables[key])
+            except Exception as exc:
+                logger.warning(
+                    f"Cannot find service for {service} - type DatabaseService - {exc}"
+                )
 
-        if self.tables:
-            for key in self.tables:
-                yield from self._parse_table_entity(key, self.tables[key])
-        if self.topics:
-            for topic in self.topics:
-                yield from self._parse_topic_entity(topic)
+        for service in self.service_connection.messagingServiceName or []:
+            try:
+                create_service = self.metadata.get_by_name(
+                    entity_type=MessagingService, fqn_search_string=service
+                )
+                for key in ENTITY_TYPES.get("Topic", []):
```

Comment:
@NiharDoshi99 since we now have `ENTITY_TYPES` as below and I am assuming this can't be modified at runtime (is that a correct assumption?) we can do a key access direclty `ENTITY_TYPES["Topic"]`

```
ENTITY_TYPES = {
     "Table": {"Table": {"db": "db", "column": "columns"}},
     "Topic": {"Topic": {"schema": "schema"}},
 }
```

Reply:
oh yes .